### PR TITLE
A simple way to create and evaluate given a  'task' in the catalog and python data structure

### PR DESCRIPTION
**Pull request by yoavkatz**
- added api methods to create and evaluate datasets from python data structures
- created simplfied qa task and qa metric catalogs
- added default templates to Task, so user does need to specifiy it
- added input checks in llm_as_judges and standard recipe

### CODE REVIEW

File: docs/docs/installation.rst [hunk 1]
```diff
@@ -4,78 +4,12 @@
...
-installed, and the versions are not compatible. To fix this issue, you should choose one of the following three options:
-* Update the locally installed Unitxt
-to the Unitxt HuggingFace dataset version
-* Uninstall the local Unitxt package (in case you don't require the access to Unitxt
-direct APIs), or 
-* Change the code to load the datasets using the direct Unitxt APIs without using the HuggingFace API.
-
+The core of Unitxt has a minimal set of requirements.
+Specific metrics, inference APIs and other assets, may require additional dependencies. 
+A clear error message on what needs to be installed is provided if these dependencies are not available when the specific asset is used.
```
Comment (elronbandel) [inline docs/docs/installation.rst:15]: A general comment, maybe worth adding `what_requirements("cards.wnli")` function?

File: prepare/templates/qa/with_context.py [hunk 2]
```diff
@@ -63,6 +64,32 @@
     overwrite=True,
 )
 
+add_to_catalog(
+    MultiReferenceTemplate(
+        instruction="Answer the question based on the information provided in the {context_type} given below.",
```
Comment (elronbandel) [inline prepare/templates/qa/with_context.py:69]:
```suggestion
        instruction="Answer the question based on the information provided in the given {context_type}.",
```
"below" assume something about the spatial order (Format Role)

File: prepare/templates/qa/with_context.py [hunk 3]
```diff
@@ -63,6 +64,32 @@
...
+        references_field="answers",
+        title_fields=["context_type"],
+    ),
+    "templates.qa.with_context",
+    overwrite=True,
+)
+
+add_to_catalog(
+    MultiReferenceTemplate(
+        instruction="Answer the question directly based on the information provided in the {context_type}. Respond with the exact phrase within the {context_type} that provides the answer.",
```
Comment (elronbandel) [inline prepare/templates/qa/with_context.py:82]: Maybe:
Extract the exact phrase from the {context_type} that directly answers the question, without any alterations.

Comment (elronbandel) [reply prepare/templates/qa/with_context.py:82]: To make it more clear..

File: docs/docs/adding_task.rst [hunk 4]
```diff
@@ -58,6 +58,31 @@ in this case, the metrics calculate the accuracy of the sum of two integers, exp
...
+   template = InputOutputTemplate(
+         input_format="How much is {num1} plus {num2}?",
+         output_format="{sum}",
+   ) 
+    
+   task = Task(
+        input_fields={"num1" : int, "num2" : int},
+        reference_fields={"sum" : int},
+        prediction_type=int,
+        metrics=[
```
[the remark this answers is not in the document]
Comment (yoavkatz) [reply docs/docs/adding_task.rst:79]: Not sure I understand. We don't have this metric implemented.

File: docs/docs/installation.rst (hunk 1 above)
Comment (yoavkatz) [reply docs/docs/installation.rst:15]: Yes. Could be nice.

**Review by elronbandel — CHANGES_REQUESTED** [review]
Mainly minor comments.

**Review by elronbandel — APPROVED** [review]

**Pull request merged**